### PR TITLE
Added variable HOST_VERSION

### PR DESCRIPTION
**Pull request by pablo-herranz**
The host_configuration test uses VERSION to determine the URL of the SUSE CA repository. This is wrong, as the VERSION field in those tests represent the VERSION of the container image, not the host version. We've added the variable HOST_VERSION to determine this.

- Related ticket: https://progress.opensuse.org/issues/xyz
- Needles: https://github.com/os-autoinst/os-autoinst-needles-opensuse/pull/xyz
- Verification runs:
  - [BCI base 15-SP5 on 15-SP5](https://openqa.suse.de/tests/12931274) :white_check_mark: 
  - [BCI base 15-SP5 on Liberty 9](https://openqa.suse.de/tests/12931275) :white_check_mark: 
  - [BCI base 15-SP5 on Leap 15.5](https://openqa.suse.de/tests/12931276) :white_check_mark:

Related to: https://gitlab.suse.de/qac/qac-openqa-yaml/-/merge_requests/1400

### CODE REVIEW

File: tests/containers/host_configuration.pm [hunk 1]
```diff
@@ -27,8 +27,18 @@ sub run {
...
     if ($host_distri =~ /sles|opensuse/) {
+        my $host_version = (get_required_var('HOST_VERSION') =~ /SP/) ? "SLES_" . (get_required_var('HOST_VERSION') =~ s/-SP/_SP/r) : get_required_var('HOST_VERSION');
         zypper_call("--quiet up", timeout => $update_timeout);
-        ensure_ca_certificates_suse_installed();
+        # Cannot use `ensure_ca_certificates_suse_installed` as it will depend
+        # on the BCI container version instead of the host
+        if (script_run('rpm -qi ca-certificates-suse') == 1) {
+            zypper_call("ar --refresh http://download.suse.de/ibs/SUSE:/CA/$host_version/SUSE:CA.repo");
+        }
+        if (is_sle_micro) {
```
Comment (grisu48) [inline tests/containers/host_configuration.pm:37]: This should be part of the if block.

File: tests/containers/host_configuration.pm [hunk 2]
```diff
@@ -27,8 +27,18 @@ sub run {
     # Update the system to get the latest released state of the hosts.
     # Check routing table is well configured
     if ($host_distri =~ /sles|opensuse/) {
+        my $host_version = (get_required_var('HOST_VERSION') =~ /SP/) ? "SLES_" . (get_required_var('HOST_VERSION') =~ s/-SP/_SP/r) : get_required_var('HOST_VERSION');
```
Comment (grisu48) [inline tests/containers/host_configuration.pm:30]:
```suggestion
        my $host_version = (get_required_var('HOST_VERSION') =~ /SP/) ? "SLE_" . (get_required_var('HOST_VERSION') =~ s/-SP/_SP/r) : get_required_var('HOST_VERSION');
```

At least if I look at http://download.suse.de/ibs/SUSE:/CA, it should be `SLE_` not `SLES_`.

Comment (pablo-herranz) [reply tests/containers/host_configuration.pm:30]: Hmmm I'm confused because it's working now with SLES...
I'll change it

**Review by ricardobranco777 — CHANGES_REQUESTED** [review]

File: tests/containers/host_configuration.pm [hunk 3]
```diff
@@ -27,8 +27,18 @@ sub run {
...
     # Check routing table is well configured
     if ($host_distri =~ /sles|opensuse/) {
+        my $host_version = (get_required_var('HOST_VERSION') =~ /SP/) ? "SLE_" . (get_required_var('HOST_VERSION') =~ s/-SP/_SP/r) : get_required_var('HOST_VERSION');
         zypper_call("--quiet up", timeout => $update_timeout);
-        ensure_ca_certificates_suse_installed();
+        # Cannot use `ensure_ca_certificates_suse_installed` as it will depend
+        # on the BCI container version instead of the host
+        if (script_run('rpm -qi ca-certificates-suse') == 1) {
+            zypper_call("ar --refresh http://download.suse.de/ibs/SUSE:/CA/$host_version/SUSE:CA.repo");
+            if (is_sle_micro) {
```
[the remark this answers is not in the document]
Comment (grisu48) [reply tests/containers/host_configuration.pm:36]: Also this is not needed here, as we're not testing BCI containers on transactional systems. You can just remove this block.

File: tests/containers/host_configuration.pm [hunk 4]
```diff
@@ -27,8 +27,18 @@ sub run {
     # Update the system to get the latest released state of the hosts.
     # Check routing table is well configured
     if ($host_distri =~ /sles|opensuse/) {
+        my $host_version = (get_required_var('HOST_VERSION') =~ /SP/) ? "SLE_" . (get_required_var('HOST_VERSION') =~ s/-SP/_SP/r) : get_required_var('HOST_VERSION');
```
[the remark this answers is not in the document]
Comment (pablo-herranz) [reply tests/containers/host_configuration.pm]: Completelly agree, very suboptimal implementation. Changed

File: tests/containers/host_configuration.pm (hunk 3 above)
[the remark this answers is not in the document]
Comment (pablo-herranz) [reply tests/containers/host_configuration.pm]: Removed

File: tests/containers/host_configuration.pm (hunk 4 above)
[the remark this answers is not in the document]
Comment (pablo-herranz) [reply tests/containers/host_configuration.pm]: Totally agree, changed

**Review by ricardobranco777 — APPROVED** [review]

**Comment by DimStar77** [comment]
* https://openqa.opensuse.org/tests/3769795 seems to be broken due to this change.

**Comment by DimStar77** [comment]
> * https://openqa.opensuse.org/tests/3769795 seems to be broken due to this change.

https://progress.opensuse.org/issues/151768

**Comment by grisu48** [comment]
https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/18234

**Pull request merged**